e2e/tests/test-extension.spec.ts

```diff
@@ -2,38 +2,110 @@ import { test, expect } from './fixtures';
-  await expect(page.locator('text="2222 4000 7000 0005"')).toBeVisible();
-  // Maestro not visible
-  await expect(page.locator('text="6771 7980 2100 0008"')).not.toBeVisible();
+// verify favourites message
+test('fav', async ({ page, extensionId }) => {
+    await page.goto(`chrome-extension://${extensionId}/panel.html`);
 
+    await expect(page.locator('text="Add favourites if you like :-)"')).toBeVisible();
 });
 
-// verify copy card number to clipboard
-test('copy', async ({ page, extensionId }) => {
+// verify copy card to clipboard
+test('copy card details', async ({ page, extensionId }) => {
+    await page.goto(`chrome-extension://${extensionId}/panel.html`);
+
+    // card number
+    let cardnumber = page.locator('text="4871 0499 9999 9910"');
+    await expect(cardnumber).toBeVisible();
+    await cardnumber.click();
+
+    let clipboard = await page.evaluate("navigator.clipboard.readText()");
+    expect(clipboard).toContain("4871 0499 9999 9910");
+
+    // cvc
+    let cvc = page.locator('text="7373"').first();
+    await expect(cvc).toBeVisible();
+    await cvc.click();
+
+    clipboard = await page.evaluate("navigator.clipboard.readText()");
+    expect(clipboard).toContain("7373");
+});
+
+// verify copy IBAN to clipboard
+test('copy IBAN details', async ({ page, extensionId }) => {
+    await page.goto(`chrome-extension://${extensionId}/panel.html`);
+
+    // IBAN
+    let iban = page.locator('text="IT60X0542811101000000123456"');
+    await expect(iban).toBeVisible();
+    await iban.click();
+
+    let clipboard = await page.evaluate("navigator.clipboard.readText()");
+    expect(clipboard).toContain("IT60X0542811101000000123456");
+
+    // name
+    let name = page.locator('text="A. Pacini"');
+    await expect(name).toBeVisible();
+    await name.click();
 
+    clipboard = await page.evaluate("navigator.clipboard.readText()");
+    expect(clipboard).toContain("A. Pacini");
+
+});
+
+// verify copy Giftcard to clipboard
+test('copy Giftcard details', async ({ page, extensionId }) => {
     await page.goto(`chrome-extension://${extensionId}/panel.html`);
-  
-    const firstRow = await page.$('tr');
-    if(firstRow == null) {
-        throw new Error("firstRow not found");
-    }
-
-    const copyLink = await firstRow.$(`a.copyLinkClick`);
-    if(copyLink == null) {
-        throw new Error("copyLink not found");
-    }
-
-    await copyLink.click(); 
-
-    // clipboard includes first card number
-    const clipboardText = await page.evaluate(() => navigator.clipboard.readText());
-    expect(clipboardText).toBe("3700 0000 0000 002");
-  
+
+    // card number
+    let cardnumber = page.locator('text="6036280000000000000"');
+    await expect(cardnumber).toBeVisible();
+    await cardnumber.click();
+
+    let clipboard = await page.evaluate("navigator.clipboard.readText()");
+    expect(clipboard).toContain("6036280000000000000");
+
+    // code
+    let code = page.locator('text="123"').first();
```

Comment:
check

Reply:
- [ ] Add workflow/github action that runs test

Reply:
The GitHub action will be created as part of #30